### PR TITLE
Add purge

### PR DESCRIPTION
**Pull request by HDegroote**
Still needs to delete the dir itself

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -742,6 +749,11 @@ module.exports = class Hypercore extends EventEmitter {
     await this.core.clear(start, end)
   }
 
+  async purge () {
+    await this._closeAllSessions()
```
Comment (mafintosh) [inline index.js:753]: pass null here so its explicit

File: lib/core.js [hunk 2]
```diff
@@ -280,6 +280,33 @@ module.exports = class Core {
...
+        const cb = (err, val) => err ? reject(err) : resolve(val)
+        this.bitfield.storage.unlink(cb)
+      }),
+      new Promise((resolve, reject) => {
+        const cb = (err, val) => err ? reject(err) : resolve(val)
+        this.blocks.storage.unlink(cb)
+      })
+    ]
+
+    await Promise.all(proms)
```
Comment (mafintosh) [inline lib/core.js:307]: just do

```js
return new Promise((resolve, reject) => {
  let missing = 4
  let error = null

  this.oplog.storage.unlink(done)
  this.tree.storage.unlink(done)
  this.bitfield.storage.unlink(done)
  this.blocks.storage.unlink(done)

  function done (err) {
    if (err) error = err
    if (--missing) return
    if (error) reject(err)
    else resolve()
  }
})
```

Comment (HDegroote) [reply lib/core.js]: Nice, fixed

File: test/purge.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  t.is(core.closed, true)
+  t.is(fs.existsSync(oplogLoc), false)
+  t.is(fs.existsSync(treeLoc), false)
+  t.is(fs.existsSync(bitfieldLoc), false)
+  t.is(fs.existsSync(dataLoc), false)
+  t.is(fs.readdirSync(dir).length, 0) // Nothing remains
+})
+
+test('purge closes all sessions', async function (t) {
+  const dir = createTmpDir()
```
Comment (mafintosh) [inline test/purge.js:35]: pass the teardown to the tmp helper in all tests no? so the test folders are gc'ed

**Pull request merged**